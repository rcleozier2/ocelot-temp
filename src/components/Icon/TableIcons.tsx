import React from "react";

const Complete = (props: any) => {
  return (
    <>
      <p>
        {props.data.completedOntime} - {props.data.completedLate}
      </p>
    </>
  );
};

const Failed = (props: any) => {
  return (
    <>
      <p>
        {props.data.failedOntime} - {props.data.failedLate}
      </p>
    </>
  );
};

const JustSend = (props: any) => {
  return (
    <>
      <p>{props.data.justSends}</p>
    </>
  );
};

export { Complete, Failed, JustSend };
