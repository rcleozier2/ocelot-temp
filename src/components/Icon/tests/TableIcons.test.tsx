import React from "react";
import { Assigned, Complete, Scheduled, Failed, JustSend } from "../TableIcons";

import renderer from "react-test-renderer";

const props = {
  data: {
    scheduled: 1,
    completed: 2,
    completedLate: 3,
    failed: 2,
    failedLate: 2,
    justSends: 5
  }
};

it("renders correctly", () => {
  const component = renderer.create(<Assigned {...props} />).toJSON();
  expect(component).toMatchSnapshot();
});

it("renders correctly", () => {
  const component = renderer.create(<Complete {...props} />).toJSON();
  expect(component).toMatchSnapshot();
});

it("renders correctly", () => {
  const component = renderer.create(<Scheduled {...props} />).toJSON();
  expect(component).toMatchSnapshot();
});

it("renders correctly", () => {
  const component = renderer.create(<Failed {...props} />).toJSON();
  expect(component).toMatchSnapshot();
});

it("renders correctly", () => {
  const component = renderer.create(<JustSend {...props} />).toJSON();
  expect(component).toMatchSnapshot();
});
