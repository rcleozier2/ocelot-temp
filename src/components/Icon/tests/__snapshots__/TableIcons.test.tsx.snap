// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `<p />`;

exports[`renders correctly 2`] = `
<p>
  2
  <span
    style={
      Object {
        "color": "red",
      }
    }
  >
    (
    3
    )
  </span>
</p>
`;

exports[`renders correctly 3`] = `
<p>
  1
</p>
`;

exports[`renders correctly 4`] = `
<p>
  2
  <span
    style={
      Object {
        "color": "red",
      }
    }
  >
    (
    2
    )
  </span>
</p>
`;

exports[`renders correctly 5`] = `
<p>
  5
</p>
`;
